### DISCUSSION TITLE
Where are Dockerfiles for hub.docker.com/r/emscripten/emsdk maintained

### DISCUSSION THREAD

**Discussion by ericoporto** [emscripten-core/emscripten#19552] [Q&A] [answered]
Hi!

I have been using the docker image from the published images on dockerhub with great success for the past years, great work publishing them!

For production builds I have a need for pinned tags, but the CI provider I am currently using for the moment (Cirrus CI) has a new policy that images cannot have layers that are non-reproducible - say, if you use an image that uses another image, the chain of images has to be complete, if one of the previous images is dropped, than the succeeding ones are not available anymore.

Unfortunately this cache policy has affected the Emsdk images (at least the specific versions I have pinned). So because of this, I would like to look into the dockerfiles so I could rebuild them for my use from scratch.

Thanks!

**Comment by sbc100** [comment, marked as answer]
Its part of the emsdk repo: https://github.com/emscripten-core/emsdk/tree/main/docker

**Reply by ericoporto** [reply]
Thank you @sbc100 !!!!
